lib/Twig/Environment.php

```diff
@@ -117,6 +119,40 @@ public function __construct(Twig_LoaderInterface $loader = null, $options = arra
     }
 
     /**
+     * Returns deprecations for templates contained in a directory.
+     *
+     * @param string $dir The directory where templates are stored
+     * @param string $ext Limit the loaded templates by extension
+     *
+     * @return An array of deprecations
+     */
+    public function getDeprecations($dir, $ext = '.twig')
+    {
+        $deprecations = array();
+        $previous = set_error_handler(function ($type, $msg) use (&$deprecations) {
+            if (E_USER_DEPRECATED === $type) {
+                $deprecations[] = $msg;
+            }
+        });
+
+        foreach (new RecursiveIteratorIterator(new RecursiveDirectoryIterator($dir), RecursiveIteratorIterator::LEAVES_ONLY) as $file) {
+            if (!preg_match('{'.preg_quote($ext).'$}', $file)) {
```

Comment:
why not using a RegexIterator for that ?

and excaping the extension to build the regex should be done outside the loop